python/ray/serve/async_goal_manager.py

```diff
@@ -8,34 +8,48 @@
 
 
 class AsyncGoalManager:
+    """
+    Class to help ServeController to manage desirable states of replica.
```

Comment:
```suggestion
    Class to help ServeController to manage desired states of replica.
```

Reply:
This comment is a leaky abstraction, the goal manager has nothing to do with managing replica states